rust_src/src/buffers.rs

```diff
@@ -890,6 +891,42 @@ pub extern "C" fn build_overlay(
     }
 }
 
+/// Delete the overlay OVERLAY from its buffer.
+#[lisp_fn]
```

Comment:
Can you not use `LispOverlayRef` here instead of explicitly converting the overlay?